common/page.go

```diff
@@ -687,34 +687,54 @@ func (p *Page) Reload(opts goja.Value) api.Response {
 		k6ext.Panic(p.ctx, "parsing reload options: %w", err)
 	}
 
-	ch, evCancelFn := createWaitForEventHandler(p.ctx,
-		p.frameManager.MainFrame(),
-		[]string{EventFrameNavigation}, func(data any) bool {
+	timeoutCtx, timeoutCancelFn := context.WithTimeout(p.ctx, parsedOpts.Timeout)
+	defer timeoutCancelFn()
+
+	ch, evCancelFn := createWaitForEventHandler(
+		timeoutCtx, p.frameManager.MainFrame(), []string{EventFrameNavigation},
+		func(data any) bool {
 			return true // Both successful and failed navigations are considered
 		},
 	)
 	defer evCancelFn() // Remove event handler
 
+	lifecycleEvtCh, lifecycleEvtCancel := createWaitForEventPredicateHandler(
+		timeoutCtx, p.frameManager.MainFrame(), []string{EventFrameAddLifecycle},
+		func(data any) bool {
+			if le, ok := data.(LifecycleEvent); ok {
+				return le == parsedOpts.WaitUntil
+			}
+			return false
+		})
+	defer lifecycleEvtCancel()
+
 	action := cdppage.Reload()
 	if err := action.Do(cdp.WithExecutor(p.ctx, p.session)); err != nil {
 		k6ext.Panic(p.ctx, "reloading page: %w", err)
 	}
 
+	wrapTimeoutError := func(err error) error {
```

Comment:
Copied, but maybe a good candidate to be it's own function?

Reply:
There is quite a bit of refactoring to be done with all these lifecycle events. I will do them in a new PR once tall these PRs are in (including moving `wrapTimeoutError` into its own function). I think that's the best strategy for now, but I will be mindful in the future to do them earlier when possible.

Reply:
No worries 🤝